### PR TITLE
add flags keeppointsandlines and ignoreoutsidegrid

### PR DESCRIPTION
**Pull request by roelarents**
# Description

* add/restore flag `keeppointsandlines` (default false)
* add flag `ignoreoutsidegrid` (default false)

## Type of change

- Improvement of existing feature

# Checklist:

- [x] I've double-checked the code in this PR myself
- [x] I've left the code better than before ([boy scout rule](https://wiki.c2.com/?BoyScoutRule))
- [x] The code is readable, comments are added that explain hard or non-obvious parts.
- [x] I've expanded/improved the (unit) tests, when applicable
- [x] I've run (unit) tests that prove my solution works
- [x] There's no sensitive information like credentials in my PR

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,18 +28,32 @@ RUN go test ./... -covermode=atomic
 
 RUN go build -v -buildvcs=true -ldflags='-s -w -linkmode auto' -a -installsuffix cgo -o /texel .
 
-# FROM scratch
-FROM golang:1.21-bullseye
+## FROM scratch
+#FROM golang:1.21-bullseye
```
Comment (kad-dirc) [inline Dockerfile:32]: Heb je dit blok comments expres laten staan? Zo ja, misschien een comment waarom?

File: pointindex/pointindex.go [hunk 2]
```diff
@@ -118,31 +118,47 @@ func (ix *PointIndex) InsertPolygon(polygon geom.Polygon) {
...
+	return fmt.Sprintf("trying to insert a coord (%v, %v) outside the grid/extent (0, %v; 0, %v)", e.deepestX, e.deepestY, e.deepestSize, e.deepestSize)
 }
 
 // InsertCoord inserts a Point by its x/y coord on the deepest level
-func (ix *PointIndex) InsertCoord(deepestX int, deepestY int) {
+func (ix *PointIndex) InsertCoord(deepestX int, deepestY int) error {
 	if deepestX < 0 || deepestY < 0 || deepestX > int(ix.deepestSize)-1 || deepestY > int(ix.deepestSize)-1 {
-		// should never happen
-		panic(fmt.Errorf("trying to insert a coord (%v, %v) outside the grid/extent (0, %v; 0, %v)", deepestX, deepestY, ix.deepestSize, ix.deepestSize))
+		return OutsideGridError{
```
Comment (kad-dirc) [inline pointindex/pointindex.go:154]: Voor mijn eigen interesse: vervang je de error hier door een struct+func?

**Review by kad-dirc — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (roelarents) [reply Dockerfile:32]: o, nee

File: pointindex/pointindex.go (hunk 2 above)
Comment (roelarents) [reply pointindex/pointindex.go:154]: een struct+func die Error() implementeert en dus een error is
https://medium.com/@vijeshomen/handling-custom-errors-in-go-a-deep-dive-into-errors-is-and-errors-as-1428c989c631

**Pull request merged**